src/pybind/mgr/dashboard/controllers/__init__.py

```diff
@@ -500,10 +509,13 @@ def proxy(self):
 
         @property
         def url(self):
+            ctrl_path = self.ctrl.get_path()
+            if ctrl_path == "/":
+                ctrl_path = ""
             if self.config['path'] is not None:
-                url = "{}{}".format(self.ctrl.get_path(), self.config['path'])
+                url = "{}{}".format(ctrl_path, self.config['path'])
             else:
-                url = "{}/{}".format(self.ctrl.get_path(), self.func.__name__)
+                url = "{}/{}".format(ctrl_path, self.func.__name__)
```

Comment:
could put
```py
url = '/'.join([ctrl_path, self.config.get('path', self.func.__name__)])
url = url.strip('/')
```

Reply:
That will not work for all cases. `ctrl_path` can be just `"/"` or `"/hello"`, and `self.config['path']` can be also just `"/"` or `"/test/ping"`. In the end we need url to be a correct path starting with a slash.

Reply:
@rjfd i see.
how about
```
url = '/'.join([ctrl_path, self.config.get('path', self.func.__name__)])
os.path.normpath(url)
```

unless it's considered to be an example of abusing `os.path.normpath()`.

Reply:
@tchaikov thanks!